src/Domain/Model/Task/TaskRead.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Domain\Model\Task;
+
+use Assert\Assert;
+
+final readonly class TaskRead
+{
+    public function __construct(
+        private int $id,
+        private string $title,
+        private bool $completed,
+    ) {
+        Assert::that($title)->notEmpty("Title can't be an empty string");
+    }
+
+    public function id(): int
+    {
+        return $this->id;
+    }
+
+    public function title(): string
+    {
+        return $this->title;
+    }
+
+    public function isCompleted(): bool
+    {
+        return $this->completed;
```

Comment:
No need for public getters as the model is immutable already through the `readonly` keyword.

Reply:
That's true. Would you never include getters though? For example, and coincidentally enough though, 
the rfc for [property hooks ](https://wiki.php.net/rfc/property-hooks) you mentioned today got one reason why using getters from the beginning can prevent breaking changes incase we need a getter later on. In this application however, it might not be a big deal because it's pretty simple, but I wonder how it would be in bigger projects.